### DISCUSSION TITLE
Coexistence with NextCloud on one server

### DISCUSSION THREAD

**Discussion by desvalets** [TriliumNext/Trilium#2099] [Q&A]
I'm running a NextCloud instance (not using a container or a virtual machine) on a standard Ubuntu 20.04 install with Apache. Nextcloud has it's own domain name and is running behind an Apache reverse proxy gateway which directs port 443 traffic to the Nextcloud backend machine. The LetsEncrypt TLS certificate is on the reverse proxy. The backend has a self-signed certificate. 
The Nextcloud installation is not saturated so I'd like to add the Trilium server to this machine. My question :  Can the Trilium notes server co-exist in this environment ? I assume I'd reach the Trilium server on the backend by directing port 8080 to the Trilium server ? Or maybe I need a second domain name to separate the traffic ? Any help is appreciated. 
PS :  Currently using Trilium on single machine.   
Regards

**Comment by Zocker1999NET** [comment]
Yes, you propably can install anything aside Nextcloud or any other webserver, it would only not be recommended if your server has not enough power. You will require a second (sub-)domain to split the traffic and avoid unnecessary problems. In theory, it would be possible to host Nextcloud on `example.com/nextcloud` and Trilium on `example.com/trilium`, but Trilium probably does not support this perfectly. But you normally, you should not be required to buy a new domain name, as you should be able to add new subdomains to it.

Assuming you "own" a domain name and your Nextcloud is currently available on `example.com` / `cloud.example.com`, you can do the following (most steps similar to your Nextcloud setup):
- setup the subdomain `trilium.example.com` or `notes.example.com` (or anything you like) at your domain registrar / DNS server
- configure an additional "virtual server" on the reverse proxy using the correct domain which redirects all traffic to 8080 (or any current free port, which will be occupied by Trilium afterwards)
- configure your Let's Encrypt client to also generate certificates for your new subdomain
- configure / start your Trilium server as you like, the wiki articles of this repository will help you (if you choose another port than 8080, ensure you instruct Trilium to do so)
- ensure, your Trilium instance is **not** available on notes.example.com:8080 (replace port). It would not be big flaw, but it prevents your reverse proxy to protect your Trilium back end in any possible way and it allows you to use Trilium without encryption. Configure your firewall to so or, if you run Trilium in Docker, change the port binding from `8080:8080` to `127.0.0.1:8080:8080`.
- your Trilium instance should be available on https://notes.example.com after reloading / restarting your reverse proxy and the Let's Encrypt client
- setup sync as described in the wiki. **Do not transfer the database manually, as Trilium might not support this**

You should be able to find step-to-step guides for each point here on the Internet, as these depend on your exact current configuration. Hope this little guide helps anyone ^^

**Reply by desvalets** [reply]
Many many thanks for this detailed answer !
